.github/workflows/docs.yml

```
@@ -0,0 +1,41 @@
+name: Documentation
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3.1.0
```

We can ping to v3 for actions that are expected to not change that much in minor and micro releases.

```suggestion
    - uses: actions/checkout@v3
```

---

Got it! So, it is a good practice to choose these versions or just ins this case of using GHA?

---

That is a very complicated question! Here is the gist of it as I understand:

1. Major versions are OK, if you expected upstream to follow SemVer, to avoid surprises.
2. Some don't choose the latest when using Major, so we are forced to use Major.Minor.Micro
3. If you are crazy about security, specially with GHA that access secrets, you would use a specific commit b/c releases can be modified!
4. If you don't care about all that and the GHA is in flux, latest main is OKish.

Most of the GHA we uses fall on 1-2 categories. We trust them, we don't want surprises with new release, give me latest Major version (or Major.Minor.Micro if you cannot give me latest Major).